torch/testing/_internal/common_distributed.py

```diff
@@ -99,6 +99,23 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+# This decorator helps avoiding initializing cuda while testing other backends
+def nccl_skip_if_lt_x_gpu(backend, x):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            if backend != "nccl":
+                return func(*args, **kwargs)
```

Comment:
This means this decorator won't skip the test even if the ci env does not contain x or more GPUs?

Reply:
correct, if only skips test if backend is nccl and we have < x GPUs

Reply:
I see. Is it possible to avoid introducing this additional decorator by having two test entries? E.g.:

```python
def _test_xyz(self, ...):
  ...

@unittest.skipIf(BACKEND == "nccl")
def test_xyz(self):
  self._test_xyz()

@unittest.skipIf(BACKEND != "nccl")
@skip_if_lt_x_gpu(2)
def test_xyz_nccl:
  self._test_xyz()

```

Reply:
If we add more testing functions, I think it's more readable with decorator rather than duplicating them and adding more helpers?

Reply:
It's up to you. :)